### PR TITLE
Fix password dialog cancel error

### PR DESCRIPTION
**Pull request by sallar**

### CODE REVIEW

**Comment by perry-mitchell** [comment]
@sallar One of the checks failed sir..

**Comment by sallar** [comment]
@perry-mitchell this too :)

File: src/shared/actions/archives.js [hunk 1]
```diff
@@ -18,25 +18,27 @@ export const removeArchive = createAction(ARCHIVES_REMOVE);
...
-          ? err.message || unknownMessage
-          : unknownMessage
-      );
-    })
-  );
+  try {
+    const archive = await showPasswordDialog(
+      password => loadWorkspace(payload, password).catch(err => {
+        const unknownMessage = 'An unknown error has occurred';
+        return Promise.reject(
```
Comment (perry-mitchell) [inline src/shared/actions/archives.js]: Why reject? You could just throw another error as you're within a Promise anyhow - it'd work the same way and be a little neater (seeing as you're using await). What about:

```javascript
if (isError(err)) {
  throw err;
} else {
  throw new Error(unknownMessage);
}
```

Comment (sallar) [inline src/shared/actions/archives.js]: @perry-mitchell this one is for the password dialog component (swal) to handle and it wants a promise. the await thingie is for the result of that :D

**Pull request merged**